spec/out_derive_spec.rb

```diff
@@ -299,6 +299,77 @@
         }
       end
 
+      context 'counter_mode true 32bit' do
+        let(:config) { %[
+          counter_mode true
+          tag rate
+          key1 foo
+          time_unit_division false
+        ]}
+        before do
+          driver.run {
+                  driver.emit({'foo'=> 4294967295}, time)
+                  driver.emit({'foo'=> 100}, time + 60)
+                  }
+        end
+        it {
+                driver.emits[1].should == ['rate', time + 60, {'foo'=> 101}]
```

Comment:
Thanks for adding test code!! But they are fail, 

```
Failures:

  1) Fluent::DeriveOutput test emit key_pattern counter_mode true 32bit should == ["rate", 1476167965, {"foo"=>101}]
     Failure/Error: driver.emits[1].should == ['rate', time + 60, {'foo'=> 101}]
       expected: ["rate", 1476167965, {"foo"=>101}]
            got: ["rate", 1476167965, {"foo"=>100}] (using ==)
     # ./spec/out_derive_spec.rb:316:in `block (5 levels) in <top (required)>'

  2) Fluent::DeriveOutput test emit key_pattern counter_mode true 64bit should == ["rate", 1476167966, {"foo"=>101}]
     Failure/Error: driver.emits[1].should == ['rate', time + 60, {'foo'=> 101}]
       expected: ["rate", 1476167966, {"foo"=>101}]
            got: ["rate", 1476167966, {"foo"=>100}] (using ==)
     # ./spec/out_derive_spec.rb:351:in `block (5 levels) in <top (required)>'
```

Could you fix the expected values? Then I will merge this.

P.S. the gem spec file `fluent-plugin-derive.gemspec` is old, so you might be not able to run test on your computer.